### ISSUE TITLE
Accessibility request from a visually impaired user."

### ISSUE THREAD

**Issue by piano777wzy** [ZDoom/gzdoom#3007]
Hello, I'm a visually impaired player. I hope GZDoom can use the NVDA screen reader to output all content. 

I'm sorry I can't register on the GZDoom community forum, so I have to post my request here. I'm a non-English speaking user and have used AI for translation.

GZDoom already supports full keyboard operation with various menu options and extensive customizable hotkey settings (including in-game movement and firing), but currently none of the content is fed back to screen readers. The world's most popular open-source screen reader NVDA has an interface called nvdaControllerClient.dll that allows third-party programs to output content to NVDA.

I apologize that I'm not a technical expert and can only describe this from a user's perspective in my own words.

There's currently a project called TobyAccessibilityMod (a .pk3 file) that allows playing Doom/Doom2 and operating many GZDoom functions when loaded. However, it still has some unsupported areas like the .wad file selection menu at startup and the console opened by pressing ` . Being a third-party mod, it might encounter game compatibility issues.

I hope to get native accessibility support - this should be achievable through nvdaControllerClient.dll.

Thank you.

Reference links:
NVDA screen reader official site (open-source and most popular):
https://www.nvaccess.org
https://github.com/nvaccess/nvda

NVDA's nvdaControllerClient.dll for third-party program interaction (detailed GitHub documentation):
https://github.com/nvaccess/nvda/tree/master/extras/controllerClient

GZDoom community's TobyAccessibilityMod project (currently my main tool):
https://www.doomworld.com/forum/topic/134280-toby-accessibility-mod-doom-for-the-blind-v80-released/

GZDoom community thread:
https://forum.zdoom.org/viewtopic.php?t=71349

**Comment by the-phinet** [comment]
Are you able to provide an example of a game with a good implementation of accessibility options for people who cannot see? It doesn't need to be open source. I would like to better understand beyond what the Toby mod has done.

**Comment by jareds** [comment]
For an example of a really good game accessibility mod see
https://github.com/Factorio-Access/FactorioAccess
I know most of this won't directly apply but it dispatches speech to the users screen reading software when playing the game.
